src/fields/valueField.ts

```diff
@@ -40,6 +41,20 @@ export interface FieldState<V> {
   revertChanges(): void;
   /** Accepts the current changed value (if any) as the original and resets dirty/touched. */
   commitChanges(): void;
+  /** Creates a new FieldState with a transformation of the value, i.e. string to int, or feet to inches. */
+  adapt<V2>(adapter: ValueAdapter<V, V2>): FieldState<V2>;
```

Comment:
Question: This got me thinking about the Trade Partner Comms where we had lead times always stored in number of days, but may show them as weeks or months as well. How would anticipate using this for dynamic conversions?

Reply:
Ah yeah :thinking: , good question--I actually don't know how that logic is done today; lazy ask, do you have a link to the current impl? It always seemed tricky to me how the UI was going to "always store days", but then somehow let the user type in months or weeks and just guess?

Hm, I guess iirc there is probably a dropdown of Weeks/Months? And so if the dropdown is Weeks, we could do `form.leadTimeInDays.adapt(daysToWeeks)`, otherwise bind to `.adapt(daysToMOnths)`?